GoCardless/Resources/PayerAuthorisation.cs

```diff
@@ -0,0 +1,507 @@
+        [JsonProperty("country_code")]
+        public string CountryCode { get; set; }
+
+        /// <summary>
+        /// For Danish customers only. The civic/company number (CPR or CVR) of
+        /// the customer. Must be supplied if the customer's bank account is
+        /// denominated in Danish krone (DKK).
+        /// </summary>
+        [JsonProperty("danish_identity_number")]
+        public string DanishIdentityNumber { get; set; }
+
+        /// <summary>
+        /// Customer's email address. Required in most cases, as this allows
+        /// GoCardless to send notifications to this customer.
+        /// </summary>
+        [JsonProperty("email")]
+        public string Email { get; set; }
+
+        /// <summary>
+        /// Customer's surname. Required unless a `company_name` is provided.
+        /// </summary>
+        [JsonProperty("family_name")]
+        public string FamilyName { get; set; }
+
+        /// <summary>
+        /// Customer's first name. Required unless a `company_name` is provided.
+        /// </summary>
+        [JsonProperty("given_name")]
+        public string GivenName { get; set; }
+
+        /// <summary>
+        /// An [IETF Language Tag](https://tools.ietf.org/html/rfc5646), used
+        /// for both language
+        /// and regional variations of our product.
+        /// 
+        /// </summary>
+        [JsonProperty("locale")]
+        public string Locale { get; set; }
+
+        /// <summary>
+        /// Key-value store of custom data. Up to 3 keys are permitted, with key
+        /// names up to 50 characters and values up to 500 characters.
+        /// </summary>
+        [JsonProperty("metadata")]
+        public IDictionary<string, string> Metadata { get; set; }
+
+        /// <summary>
+        /// The customer's postal code.
+        /// </summary>
+        [JsonProperty("postal_code")]
+        public string PostalCode { get; set; }
+
+        /// <summary>
+        /// The customer's address region, county or department. For US
+        /// customers a 2 letter
+        /// [ISO3166-2:US](https://en.wikipedia.org/wiki/ISO_3166-2:US) state
+        /// code is required (e.g. `CA` for California).
+        /// </summary>
+        [JsonProperty("region")]
+        public string Region { get; set; }
+
+        /// <summary>
+        /// For Swedish customers only. The civic/company number (personnummer,
+        /// samordningsnummer, or organisationsnummer) of the customer. Must be
+        /// supplied if the customer's bank account is denominated in Swedish
+        /// krona (SEK). This field cannot be changed once it has been set.
+        /// </summary>
+        [JsonProperty("swedish_identity_number")]
+        public string SwedishIdentityNumber { get; set; }
+    }
+    
+    /// <summary>
+    /// An array of fields which are missing and is required to set up the
+    /// mandate.
+    /// </summary>
+    public class PayerAuthorisationIncompleteField
```

Comment:
Why is this in singular?

Reply:
I think singular is fine here as its plural in where it's used:

```c#
public List<PayerAuthorisationIncompleteField> IncompleteFields { get; set; }
```